packages/safe-ds-lang/src/language/generation/safe-ds-python-generator.ts

```diff
@@ -634,27 +638,29 @@ export class SafeDsPythonGenerator {
             return traceToNode(expression)(frame.getUniqueLambdaBlockName(expression));
         } else if (isSdsCall(expression)) {
             const callable = this.nodeMapper.callToCallable(expression);
-            if (isSdsFunction(callable)) {
-                const pythonCall = this.builtinAnnotations.getPythonCall(callable);
-                if (pythonCall) {
+            const sortedArgs = this.sortArguments(expression.argumentList.arguments);
+            // Memoize constructor or function call
+            if (isSdsFunction(callable) || isSdsClass(callable)) {
+                if (isSdsFunction(callable)) {
+                    const pythonCall = this.builtinAnnotations.getPythonCall(callable);
+                    if (pythonCall) {
+                        let thisParam: CompositeGeneratorNode | undefined = undefined;
+                        if (isSdsMemberAccess(expression.receiver)) {
+                            thisParam = this.generateExpression(expression.receiver.receiver, frame);
+                        }
+                        const argumentsMap = this.getArgumentsMap(expression.argumentList.arguments, frame);
```

Comment:
Should use `getArguments(expression)`. It's better to decouple the code as much as possible from the abstract syntax, so it's easier to change it.